demos/coreMQTT/mqtt_demo_basic_tls.c

```diff
@@ -0,0 +1,1005 @@
+    xTransport.send = SecureSocketsTransport_Send;
+    xTransport.recv = SecureSocketsTransport_Recv;
+
+    /* Initialize MQTT library. */
+    xResult = MQTT_Init( pxMQTTContext, &xTransport, prvGetTimeMs, prvEventCallback, &xBuffer );
+
+    if( xResult == MQTTSuccess )
+    {
+        /* Some fields are not used in this demo so start with everything at 0. */
+        ( void ) memset( ( void * ) &xConnectInfo, 0x00, sizeof( xConnectInfo ) );
+
+        /* Start with a clean session i.e. direct the MQTT broker to discard any
+         * previous session data. Also, establishing a connection with clean session
+         * will ensure that the broker does not store any data when this client
+         * gets disconnected. */
+        xConnectInfo.cleanSession = true;
+
+        /* The client identifier is used to uniquely identify this MQTT client to
+         * the MQTT broker. In a production device the identifier can be something
+         * unique, such as a device serial number. */
+        xConnectInfo.pClientIdentifier = democonfigCLIENT_IDENTIFIER;
+        xConnectInfo.clientIdentifierLength = ( uint16_t ) strlen( democonfigCLIENT_IDENTIFIER );
+
+        /* Set MQTT keep-alive period. If the application does not send packets
+         * at an interval less than the keep-alive period, the MQTT library will
+         * send PINGREQ packets. */
+        xConnectInfo.keepAliveSeconds = mqttexampleKEEP_ALIVE_TIMEOUT_SECONDS;
+
+        /* Send MQTT CONNECT packet to broker. LWT is not used in this demo, so it
+         * is passed as NULL. */
+        xResult = MQTT_Connect( pxMQTTContext,
+                                &xConnectInfo,
+                                NULL,
+                                mqttexampleCONNACK_RECV_TIMEOUT_MS,
+                                &xSessionPresent );
+    }
+    else
+    {
+        LogError( ( "MQTT_Init() failed with status %s.",
+                    MQTT_Status_strerror( xResult ) ) );
+    }
+
+    if( xResult != MQTTSuccess )
+    {
+        xReturnStatus = pdFAIL;
+        LogError( ( "MQTT_Connect failed with status %s.",
+                    MQTT_Status_strerror( xResult ) ) );
+    }
+
+    /* Successfully established and MQTT connection with the broker. */
+    LogInfo( ( "An MQTT connection is established with %s.", democonfigMQTT_BROKER_ENDPOINT ) );
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+static void prvUpdateSubAckStatus( MQTTPacketInfo_t * pxPacketInfo )
+{
+    MQTTStatus_t xResult = MQTTSuccess;
+    uint8_t * pucPayload = NULL;
+    size_t ulSize = 0;
+    uint32_t ulTopicCount = 0U;
+
+    xResult = MQTT_GetSubAckStatusCodes( pxPacketInfo, &pucPayload, &ulSize );
+
+    /* MQTT_GetSubAckStatusCodes always returns success if called with packet info
+     * from the event callback and non-NULL parameters. */
+    configASSERT( xResult == MQTTSuccess );
+
+    for( ulTopicCount = 0; ulTopicCount < ulSize; ulTopicCount++ )
```

Comment:
Good to assert that ulSize == mqttexampleTOPIC_COUNT and the loop should be updated to the following to avoid out-of-bound write:

```c
for( ulTopicCount = 0; ulTopicCount < mqttexampleTOPIC_COUNT; ulTopicCount++ )
```

Reply:
Updated.